packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -353,10 +369,26 @@ export function createRootHooks<
     const optsRef = React.useRef<typeof opts>(opts);
     optsRef.current = opts;
```

Comment:
Isn't it the case that this is bad and not compatible with concurrent react, suspense, and RSC/SSR?

I thought we should always put these setters in a useEffect?

Reply:
Not sure, never seen an issue related to it 😬

Reply:
Because a render can be thrown away at any moment, setting refs during render is discouraged as they could end up with an invalid value, effects run after a render is committed. In practice probably not a big issue but I'm not an expert beyond that description